tap_klaviyo_custom/streams.py

```diff
@@ -1,46 +1,214 @@
+        request_data = self.prepare_request_payload(context, next_page_token)
+        headers = self.http_headers
+
+        request = cast(
+            requests.PreparedRequest,
+            self.requests_session.prepare_request(
+                requests.Request(
+                    method=http_method,
+                    url=url,
+                    params=params,
+                    headers=headers,
+                    json=request_data,
+                )
+            ),
+        )
+        return request
+
+    def get_url(self, context: Optional[dict]) -> str:
+        """Return a URL, optionally targeted to a specific partition or context.
+
+        Developers override this method to perform dynamic URL generation.
+        """
+        url = self.url_base+self.path
+
+        return url
+
+    def request_records(self, context: Optional[dict]) -> Iterable[dict]:
+        """Request records from REST endpoint(s), returning response records.
+
+        If pagination is detected, pages will be recursed automatically.
+        """
+        next_page_token: Any = None
+        finished = False
+        while not finished:
+            prepared_request = self.prepare_request(
+                context, next_page_token=next_page_token
+            )
+            resp = self._request_with_backoff(prepared_request, context)
+            for row in self.parse_response(resp):
+                yield row
+            previous_token = copy.deepcopy(next_page_token)
+            next_page_token = self.get_next_page_token(
+                response=resp, previous_token=previous_token
+            )
+            if next_page_token and next_page_token == previous_token:
+                raise RuntimeError(
+                    f"Loop detected in pagination. "
+                    f"Pagination token {next_page_token} is identical to prior token."
+                )
+            # Cycle until get_next_page_token() no longer returns a value
+            finished = not next_page_token
+
+
+    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        for row in self.request_records(context):
+            row = self.post_process(row, context)
+            yield row
+
+
+
+class ListMembersStream(RESTStream):
     """Define custom stream."""
-    name = "groups"
-    path = "/groups"
-    primary_keys = ["id"]
-    replication_key = "modified"
-    schema = th.PropertiesList(
-        th.Property("name", th.StringType),
-        th.Property("id", th.StringType),
-        th.Property("modified", th.DateTimeType),
-    ).to_dict()
+    name = "list_members"
+    path = "group/{list_id}/members/all"
+    primary_keys = ["email"]
+    replication_key = None
+    schema_filepath = SCHEMAS_DIR / "list_members.json"
+
+    url_base = "https://a.klaviyo.com/api/v2/"
```

Comment:
The same url is repeated in both the streams could we make that a global variable or define it in one place?

Reply:
You're right. Would it be a better idea to add it to the config file and refer to it that way or to add it outside the stream classes, but still inside streams.py as a global variable?